pkg/handlers/v1/nexposevulnnotifier_test.go

```diff
@@ -116,6 +117,7 @@ func TestNexposeVulnNotificationHandlerError(t *testing.T) {
 func TestNexposeVulnNotificationHandlerWithAssetsAndErrors(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
+	mockLogger := NewMockLogger(mockCtrl)
```

Comment:
suggestion: It might be better to use a stub for the logger (like the stat client) rather than a mock.

Mocks work really well when the thing we're mocking is important to the correct functioning of the code under test. For example, we mock the asset fetcher because the return values from using it are critical to how the handler works. Depending on the return values, the handler will enter one of multiple branches of code. Using a mock enables us to have fine grained control over which code branches are active during the test.

For cases where the thing we're mocking is _not_ important to the correct functioning of the code then a mock might be too much. A mock results in test failure for any unexpected call which means each test will need to set expectations for _all_ calls to a thing that is irrelevant to the code under test. In these cases it's probably better to use a stub and bypass the state management of a mock.

With respect to logging/stats, an exception would be when a log or stat _is_ important to us and we want to guarantee that it is emitted. For example, if the only indicator of a condition is a log that we rely on to operate a system then we'd likely benefit from testing that the log is emitted during some condition. However, I'd also recommend having a test specifically for that rather than using a mock in all tests.

Reply:
> A mock results in test failure for any unexpected call which means each test will need to set expectations for all calls to a thing that is irrelevant to the code under test

I thought this made the test better because if I'm testing an error case, then I want to make sure that an error is logged. How would the test be different if I'm testing that a log is emitted during an error case?

Reply:
That's the kind of exception I meant to describe as being a good reason to use a mock logger. Generally, an error log isn't the only indicator of an issue but you have an exceptional case. If you're using the log mock as an equivalent to `require.NoError()` for the tests then that's OK. I think I failed to identify that this is what you were doing.

Reply:
ah, right. This is a result of the handler not returning anything, even an error. I'll leave it as is then.